File: Java/src/main/java/jakopec/Start.java
package jakopec;


import jakopec.model.*;
import jakopec.pomocno.HibernateUtil;
import org.hibernate.Session;

import java.util.ArrayList;
import java.util.List;

public class Start {

    private  Session s ;
    private List<Partija> partije;
    private Igrac igrac1;
    private Igrac igrac2;
    private Igrac igrac3;
    private Igrac igrac4;
    private Lokacija lokacija;

    public Start() {
        s = HibernateUtil.getSession();
        insert(); // prije izvođenja u hibernate.cfg.xml svojstvo hbm2ddl.auto postaviti na create
        select();
        //update();  // prije izvođenja u hibernate.cfg.xml svojstvo hbm2ddl.auto postaviti na update
        //delete();  // prije izvođenja u hibernate.cfg.xml svojstvo hbm2ddl.auto postaviti na update
    }

    private void select(){
        // podaci iz JAVA objekata kreniranih ručno
        for (Partija partija: partije) {
            System.out.println(partija);
        }

        // Podaci iz baze
        List<Partija> partije = s.createQuery(
                "from Partija",Partija.class).list();
        for (Partija partija: partije) {
            System.out.println(partija);
        }
    }

    private void update(){
        List<Partija> partije = s.createQuery("from Partija", Partija.class).list();
        var mjesanje = partije.get(0).getMjesanja().get(0);
        mjesanje.setBelot(true);
        s.beginTransaction();
        s.persist(mjesanje);
        s.getTransaction().commit();
    }

    private void delete(){
        List<Partija> partije = s.createQuery("from Partija", Partija.class).list();
        var mjesanje = partije.get(0).getMjesanja().get(0);
        s.beginTransaction();
        s.remove(mjesanje);
        s.getTransaction().commit();
    }

    private void insert(){
        s.beginTransaction();
        partije = new ArrayList<>();
        igrac1 = kreirajIgraca1();
        s.persist(igrac1);
        igrac2 = kreirajIgraca2();
        s.persist(igrac2);
        igrac3 = new Igrac(null, "Marija", "Zimska", "https://picsum.photos/200", Spol.ZENSKO.getId());
        s.persist(igrac3);
        igrac4 = new Igrac(null, "Anita", "Račman", "https://picsum.photos/200", Spol.ZENSKO.getId());
        s.persist(igrac4);
        lokacija = kreirajLokaciju();
        s.persist(lokacija);
        kreirajPartijuDvaIgraca();
        kreirajPartijuTriIgraca();
        kreirajPartijuDvaPara();
        s.getTransaction().commit();
    }

    private void kreirajPartijuDvaIgraca() {
        PartijaDvaIgraca partija = new PartijaDvaIgraca();
        partija.setDoKolikoSeIgra(501);
        partija.setLokacija(lokacija);
        partija.setUnosi(igrac1);
        List<Igrac> igraci = new ArrayList<>();
        igraci.add(igrac1);
        igraci.add(igrac2);
        partija.setIgraci(igraci);
        partija.setMjesanja(kreirajMjesanjaDvaIgraca(partija));
        s.persist(partija);
        partije.add(partija);
    }

    private List<Mjesanje> kreirajMjesanjaDvaIgraca(Partija p) {
        List<Mjesanje> mjesanja = new ArrayList<>();
        MjesanjeDvaUnosa m = new MjesanjeDvaUnosa();
        m.setBodovaPrviUnos(10);
        m.setBodovaDrugiUnos(152);
        m.setZvanjePrviUnos(0);
        m.setZvanjeDrugiUnos(20);
        m.setPartija(p);
        s.persist(m);
        mjesanja.add(m);
        m = new MjesanjeDvaUnosa();
        m.setBodovaPrviUnos(152);
        m.setBodovaDrugiUnos(10);
        m.setZvanjePrviUnos(0);
        m.setZvanjeDrugiUnos(20);
        m.setStiglja(true);
        m.setPartija(p);
        s.persist(m);
        mjesanja.add(m);
        return mjesanja;
    }


    private void kreirajPartijuTriIgraca() {
        PartijaTriIgraca partija = new PartijaTriIgraca();
        partija.setDoKolikoSeIgra(501);
        partija.setLokacija(lokacija);
        partija.setUnosi(igrac1);
        List<Igrac> igraci = new ArrayList<>();
        igraci.add(igrac1);
        igraci.add(igrac2);
        igraci.add(igrac3);
        partija.setIgraci(igraci);
        partija.setMjesanja(kreirajMjesanjaTriIgraca(partija));
        s.persist(partija);
        partije.add(partija);
    }

    private List<Mjesanje> kreirajMjesanjaTriIgraca(Partija p) {
        List<Mjesanje> mjesanja = new ArrayList<>();
        MjesanjeTriUnosa m = new MjesanjeTriUnosa();
        m.setBodovaPrviUnos(10);
        m.setBodovaDrugiUnos(76);
        m.setBodovaTreciUnos(76);
        m.setZvanjePrviUnos(0);
        m.setZvanjeDrugiUnos(20);
        m.setZvanjeTreciUnos(0);
        m.setPartija(p);
        s.persist(m);
        mjesanja.add(m);
        for(int i=0;i<5;i++) {
            m = new MjesanjeTriUnosa();
            m.setBodovaPrviUnos(10);
            m.setBodovaDrugiUnos(76);
            m.setBodovaTreciUnos(76);
            m.setZvanjePrviUnos(0);
            m.setZvanjeDrugiUnos(20);
            m.setZvanjeTreciUnos(0);
            m.setPartija(p);
            s.persist(m);
            mjesanja.add(m);
        }
        return mjesanja;
    }

    private void kreirajPartijuDvaPara() {
        PartijaDvaPara partija = new PartijaDvaPara();
        partija.setDoKolikoSeIgra(501);
        partija.setLokacija(lokacija);
        partija.setUnosi(igrac1);
        List<Igrac> igraci = new ArrayList<>();
        igraci.add(igrac1);
        igraci.add(igrac2);
        igraci.add(igrac3);
        igraci.add(igrac4);
        partija.setIgraci(igraci);
        partija.setMjesanja(kreirajMjesanjaDvaPara(partija));
        s.persist(partija);
        partije.add(partija);
    }

    private List<Mjesanje> kreirajMjesanjaDvaPara(Partija p) {
        List<Mjesanje> mjesanja = new ArrayList<>();
        MjesanjeDvaUnosa m = new MjesanjeDvaUnosa();
        m.setBodovaPrviUnos(10);
        m.setBodovaDrugiUnos(152);
        m.setZvanjePrviUnos(0);
        m.setZvanjeDrugiUnos(20);
        m.setPartija(p);
        s.persist(m);
        mjesanja.add(m);
        m = new MjesanjeDvaUnosa();
        m.setBodovaPrviUnos(152);
        m.setBodovaDrugiUnos(10);
        m.setZvanjePrviUnos(0);
        m.setZvanjeDrugiUnos(20);
        m.setStiglja(true);
        m.setPartija(p);
        s.persist(m);
        mjesanja.add(m);
        return mjesanja;

    }

    private Lokacija kreirajLokaciju() {
        Lokacija l = new Lokacija();
        l.setNaziv("Caffe Bar Peppermint");
        l.setLatitude(45.5605825);
        l.setLongitude(18.6098766);
        return l;
    }

    private Igrac kreirajIgraca1() {
        Igrac i = new Igrac();
        i.setIme("Tomislav");
        i.setPrezime("Jakopec");
        i.setUrlSlika("https://picsum.photos/200");
        i.setSpol(Spol.MUSKO.getId());
        return i;
    }

    private Igrac kreirajIgraca2() {
        Igrac i = new Igrac();
        i.setIme("Marijan");
        i.setPrezime("Zidar");
        i.setUrlSlika("https://picsum.photos/200");
        i.setSpol(Spol.MUSKO.getId());
        return i;
    }

    public static void main(String[] args) {
        new Start();
    }

}

File: PHP/Start.php
<?php

use Doctrine\ORM\EntityManager;
use Doctrine\ORM\ORMSetup;
use Jakopec\Igrac;
use Jakopec\Lokacija;
use Jakopec\MjesanjeDvaUnosa;
use Jakopec\MjesanjeTriUnosa;
use Jakopec\PartijaDvaIgraca;
use Jakopec\PartijaDvaPara;
use Jakopec\PartijaTriIgraca;

class Start
{
    private $entityManager;
    private $lokacija;
    private $igrac1;
    private $igrac2;
    private $igrac3;
    private $igrac4;
    private $partije;

    /**
     * Start constructor.
     */
    public function __construct()
    {

        $this->defineDoctrine();

        $this->partije = [];
        $this->ucitajPartije();
        $this->insert();

        $this->select();


    }

    private function select()
    {
        $dql = $this->entityManager->createQuery('SELECT p FROM Jakopec\Partija p');
        $partije = $dql->getResult();
        foreach ($partije as $partija) {
            echo $partija . PHP_EOL;
        }
    }

    private function insert()
    {
        foreach ($this->partije as $partija) {
            $this->entityManager->persist($partija);
        }
        $this->entityManager->flush();
    }

    private function ucitajPartije()
    {
        $this->lokacija = new Lokacija();
        $this->lokacija->setNaziv('Caffe Bar Peppermint');
        $this->lokacija->setLongitude(18.6098766);
        $this->lokacija->setLatitude(45.5605825);
        $this->entityManager->persist($this->lokacija);

        $this->igrac1 = new Igrac();
        $this->igrac1->setIme('Tomislav');
        $this->igrac1->setPrezime('Jakopec');
        $this->entityManager->persist($this->igrac1);

        $this->igrac2 = new Igrac();
        $this->igrac2->setIme('Marijan');
        $this->igrac2->setPrezime('Zidar');
        $this->entityManager->persist($this->igrac2);

        $this->igrac3 = new Igrac();
        $this->igrac3->setIme('Marija');
        $this->igrac3->setPrezime('Zimska');
        $this->entityManager->persist($this->igrac3);

        $this->igrac4 = new Igrac();
        $this->igrac4->setIme('Anita');
        $this->igrac4->setPrezime('Račman');
        $this->entityManager->persist($this->igrac4);

        $this->kreirajPartijuDvaIgraca();

        $this->kreirajPartijuTriIgraca();

        $this->kreirajPartijuDvaPara();

    }

    private function kreirajPartijuDvaIgraca()
    {
        $p = new PartijaDvaIgraca();
        $p->setDoKolikoSeIgra(501);
        $p->setLokacija($this->lokacija);
        $p->setUnosi($this->igrac1);
        $igraci = [];
        $igraci[] = $this->igrac1;
        $igraci[] = $this->igrac2;
        $p->setIgraci($igraci);
        $mjesanja = [];

        $m = new MjesanjeDvaUnosa();
        $m->setPartija($p);
        $m->setBodovaPrviUnos(10);
        $m->setBodovaDrugiUnos(152);
        $m->setZvanjePrviUnos(0);
        $m->setZvanjeDrugiUnos(20);
        $mjesanja[] = $m;

        $m = new MjesanjeDvaUnosa();
        $m->setPartija($p);
        $m->setBodovaPrviUnos(152);
        $m->setBodovaDrugiUnos(10);
        $m->setZvanjePrviUnos(0);
        $m->setZvanjeDrugiUnos(20);
        $mjesanja[] = $m;


        $p->setMjesanja($mjesanja);
        $this->entityManager->persist($p);
        $this->partije[] = $p;
    }

    private function kreirajPartijuTriIgraca()
    {
        $p = new PartijaTriIgraca();
        $p->setDoKolikoSeIgra(501);
        $p->setLokacija($this->lokacija);
        $p->setUnosi($this->igrac1);
        $igraci = [];
        $igraci[] = $this->igrac1;
        $igraci[] = $this->igrac2;
        $igraci[] = $this->igrac3;
        $p->setIgraci($igraci);
        $mjesanja = [];

        $m = new MjesanjeTriUnosa();
        $m->setPartija($p);
        $m->setBodovaPrviUnos(10);
        $m->setBodovaDrugiUnos(76);
        $m->setBodovaTreciUnos(76);
        $m->setZvanjePrviUnos(0);
        $m->setZvanjeDrugiUnos(20);
        $m->setZvanjeTreciUnos(0);

        $mjesanja[] = $m;

        for ($i = 0; $i < 5; $i++) {
            $m = new MjesanjeTriUnosa();
            $m->setPartija($p);
            $m->setBodovaPrviUnos(10);
            $m->setBodovaDrugiUnos(76);
            $m->setBodovaTreciUnos(76);
            $m->setZvanjePrviUnos(0);
            $m->setZvanjeDrugiUnos(20);
            $m->setZvanjeTreciUnos(0);
            $mjesanja[] = $m;
        }

        $p->setMjesanja($mjesanja);
        $this->entityManager->persist($p);
        $this->partije[] = $p;
    }

    private function kreirajPartijuDvaPara()
    {
        $p = new PartijaDvaPara();
        $p->setDoKolikoSeIgra(501);
        $p->setLokacija($this->lokacija);
        $p->setUnosi($this->igrac1);
        $igraci = [];
        $igraci[] = $this->igrac1;
        $igraci[] = $this->igrac2;
        $igraci[] = $this->igrac3;
        $igraci[] = $this->igrac4;
        $p->setIgraci($igraci);
        $mjesanja = [];

        $m = new MjesanjeDvaUnosa();
        $m->setPartija($p);
        $m->setBodovaPrviUnos(10);
        $m->setBodovaDrugiUnos(152);
        $m->setZvanjePrviUnos(0);
        $m->setZvanjeDrugiUnos(20);
        $mjesanja[] = $m;

        $m = new MjesanjeDvaUnosa();
        $m->setPartija($p);
        $m->setBodovaPrviUnos(152);
        $m->setBodovaDrugiUnos(10);
        $m->setZvanjePrviUnos(0);
        $m->setZvanjeDrugiUnos(20);
        $mjesanja[] = $m;


        $p->setMjesanja($mjesanja);
        $this->entityManager->persist($p);
        $this->partije[] = $p;
    }

    // ovo je inače u bootstrap.php
    private function defineDoctrine()
    {

        $config = ORMSetup::createAnnotationMetadataConfiguration(array(__DIR__ . "/src"), true, null, null);

        // database configuration parameters
        // grant all privileges on orm_php.* to 'orm'@'localhost' identified by 'orm';
        // create database orm_php character set utf8mb4 collate utf8mb4_general_ci;
        $conn = array(
            'dbname' => 'orm_php',
            'user' => 'orm',
            'password' => 'orm',
            'host' => 'localhost',
            'driver' => 'pdo_mysql',
        );
        try {
            $this->entityManager = EntityManager::create($conn, $config);
        } catch (\Doctrine\ORM\Exception\ORMException $e) {
            print_r($e);
        }
    }

}

